pep-0630.rst

```diff
@@ -321,10 +328,26 @@ the Python level: for example, you can't set ``str.myattribute = 123``.
    that shares any Python objects across interpreters implicitly depends
    on CPython's current, process-wide GIL.
 
-An alternative to static types is *heap-allocated types*, or heap types
+Because they are immutable and process-global, static types cannot access
+“their” module state.
+If any method of such a type requires access to module state,
+the type must be converted to a *heap-allocated type*, or *heap type*
 for short. These correspond more closely to classes created by Python’s
 ``class`` statement.
 
+For new modules, using heap types by default is a good rule of thumb.
+
+Existing classes can be converted to heap ones, but note that
+the heap type API was not designed for “lossless” conversion
+from static types. Unlike static types, heap type objects are mutable.
+Also, when rewriting the class definition in a new API,
+you are likely to unintentionally change a few details (e.g. pickle-ability
```

Comment:
I would reconsider this wording. _Take care not to_ IMO sounds nicer than _You are likely to_.

Reply:
Also, is the pickle issue still an issue? Wasn't that fixed by Serhiy long ago? The bpo issue that mentioned this issue is closed, marked as resolved.

Reply:
To get back to the original comment, I think changing the behavior is a fact of life rather than something to avoid. At least right now.
(It's different in the stdlib.)

Reply:
> OTOH, there are unexplored issues that stdlib doesn't run into but other libraries do, e.g. ones involving complex inheritance chains or metaclasses.
> 
> To keep the PEP general, I don't want to include the specific fixes here.

Sounds good to me.